Response Compression Middleware topic updates for 2.0

guardrex opened the pull request:
Fixes #4044 
[review URL](https://review.docs.microsoft.com/en-us/aspnet/core/performance/response-compression?branch=pr-en-us-4045&tabs=aspnetcore2x)

dnfclas commented:
@guardrex,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

guardrex commented:
@Rick-Anderson @tdykstra

Notes:
* Samples: layout spec, tabs, 1.x.y updated, 2.x 😵 Nutjob:tm: 😵 startup
* Cleaned up the text; the usual stuff
* 🥀 Table Beautification 🌷
* Added technology labeling "ASP.NET Core" to title and **\<h1>**
* Updated for new features of 2.0
* Note that GZip, GZIP, and gzip are all in use; the inventor has it as "gzip." It can be seen that way at http://www.gzip.org/, so that's what I go with here.

*It's off of WIP and ready for* :eyes:!

guardrex commented:
Hold for a sec ... I want to check on one more thing before review.

Rick-Anderson commented on aspnetcore/performance/response-compression.md:
```diff
@@ -1,97 +1,114 @@
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/response-compression/samples)
 
-Network bandwidth is a limited resource. If you can reduce response payload sizes and thus send less data to clients, you can usually increase the responsiveness of your application, sometimes dramatically. One way to reduce payload sizes is to compress an application's responses.
+Network bandwidth is a limited resource. If you can reduce response payload sizes and thus send less data to clients, you can usually increase the responsiveness of your app, often dramatically. One way to reduce payload sizes is to compress an app's responses.
```
Reducing response ~payload~ size ~and thus send less data to clients,you can~  usually increase the responsiveness of an app, often dramatically

guardrex replied:
> Reducing response size ...

... seems a little awkward just b/c using "response" as an adjective seems harder to grok. Let's try on the next commit with ...

> Reducing the size of the response ...

Rick-Anderson replied:
Agreed, it was a quick off the top of my head suggestion. They're all suggestions.

Rick-Anderson commented on aspnetcore/performance/response-compression.md:
```diff
@@ -1,97 +1,114 @@
 
 ## When to use Response Compression Middleware
-Use Response Compression Middleware when you're unable to use the [Dynamic Compression module](https://www.iis.net/overview/reliability/dynamiccachingandcompression) in IIS, the [Apache mod_deflate module](http://httpd.apache.org/docs/current/mod/mod_deflate.html), [NGINX Compression and Decompression](https://www.nginx.com/resources/admin-guide/compression-and-decompression/), or your applicatio ...
+Use Response Compression Middleware when you're unable to use the [Dynamic Compression module](https://www.iis.net/overview/reliability/dynamiccachingandcompression) in IIS, the [Apache mod_deflate module](http://httpd.apache.org/docs/current/mod/mod_deflate.html), [NGINX Compression and Decompression](https://www.nginx.com/resources/admin-guide/compression-and-decompression/), or your app is hos ...
```
Use Response Compression Middleware when you're unable to use the  the following:

Rick-Anderson replied:
Or something like that to make a bulleted list. Machine lang translators don't like long sentences, either do I.

guardrex replied:
Much, much, *MUCH* better.

Rick-Anderson commented on aspnetcore/performance/response-compression.md:
```diff
@@ -1,97 +1,114 @@
 ## Response compression
-Usually, any response not natively compressed can benefit from response compression. Responses not natively compressed typically include: CSS, JavaScript, HTML, XML, and JSON. You shouldn't compress natively compressed assets, such as PNG files. If you attempt to further compress a natively compressed response, any small additional reduction in size and transmission time will likely be overshadow ...
+Usually, any response not natively compressed can benefit from response compression. Responses not natively compressed typically include: CSS, JavaScript, HTML, XML, and JSON. You shouldn't compress natively compressed assets, such as PNG files. If you attempt to further compress a natively compressed response, any small additional reduction in size and transmission time will likely be overshadow ...
 
```
>You shouldn't
Do not 
~also~  consider that a banned word, also, also should also be used very rarely

Don't compress files smaller than about 150-1000 bytes (depending on the file's content and the efficiency of compression). The overhead of compressing small files may produce a compressed file larger than the uncompressed file .

Rick-Anderson commented on aspnetcore/performance/response-compression.md:
```diff
@@ -1,97 +1,114 @@
+| `Vary`             | When sent by the server with a value of `Accept-Encoding` to clients and proxies, the `Vary` header indicates to the client or proxy that it should cache (vary) responses based on the value of the `Accept-Encoding` header of the request. The result of returning content with the `Vary: Accept-Encoding` header is that both compressed and uncompressed responses are cached sepa ...
 
-You can explore the features of the Response Compression Middleware with the [sample application](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/response-compression/sample). The sample illustrates the compression of application responses using Gzip and custom compression providers. It also shows you how to add a MIME type to the default list of MIME types for compression.
+You can explore the features of the Response Compression Middleware with the [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/response-compression/samples). The sample illustrates the compression of app responses using gzip and custom compression providers. It also shows you how to add a MIME type to the default list of MIME types for compression.
```
The sample illustrates:

* list

Rick-Anderson commented on aspnetcore/performance/response-compression.md:
```diff
@@ -1,97 +1,114 @@
 
-[!code-csharp[Main](response-compression/sample/Startup.cs?name=snippet3)]
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
```
The following code ...

guardrex commented:
@Rick-Anderson Got it. :+1:

guardrex commented:
@Rick-Anderson Stand-by a sec. U made a comment on the other one ('u don't need this package if you use the 'All' package) that also applies over here. I'll come back and fix this one in a few minutes. Hold the launch at T-00:00:02 seconds plz. :rocket:

guardrex commented:
@Rick-Anderson kk ... ready now.

Rick-Anderson commented:
Nice work. Thanks!
